Added docu for @Common.FieldControl validation (#1474)
cap-js/docs: guides/providing-services.md

### `@Common.FieldControl` 
{#common-fieldcontrol}

> The input validation for `@Common.FieldControl: #Mandatory` and `@Common.FieldControl: #ReadOnly` is done from the CAP runtimes automatically.

Author: @johannes-vogel I think it is better to have the whole annotation here, not only the values. It will be helpful for the customer to see the whole syntax, as we don't have other documentation for this.

> The input validation for `@Common.FieldControl: #Mandatory` and `@Common.FieldControl: #ReadOnly` is done from the CAP runtimes automatically.

Reviewer: @renejeglinsky would you rather combine the values of the annotation or is that too complex? I was thinking wether we can avoid the repitition. I'm also fine with the current sentence

> The input validation for `@Common.FieldControl: #Mandatory` and `@Common.FieldControl: #ReadOnly` is done from the CAP runtimes automatically.

Reviewer 2: I was thinking if we could just use `@readonly` and `@mandatory` in this introductory sentence, as this is what you would usually use in your model, right? And then the importance is on the note, that says if use any other value for a `@Common.FieldControl` annotation, you need to validate it yourself. WDYT?

> The input validation for `@Common.FieldControl: #Mandatory` and `@Common.FieldControl: #ReadOnly` is done from the CAP runtimes automatically.

Reviewer: This is especially about the `@Common.FieldControl` annotation as this can be added to the model as well wihtout adding @readonly. We also do validate, but only static values.

> The input validation for `@Common.FieldControl: #Mandatory` and `@Common.FieldControl: #ReadOnly` is done from the CAP runtimes automatically.

Reviewer 2: Is `@readonly` a shortcut for `@Common.FieldControl: #ReadOnly`? That's what I assumed but now I don't anymore 😄

> The input validation for `@Common.FieldControl: #Mandatory` and `@Common.FieldControl: #ReadOnly` is done from the CAP runtimes automatically.

Reviewer: No, it's not a shortcut. `@readonly` is a CAP annotation, `@Common.FieldControl: #ReadOnly` is an OData annotation. The runtime supports both.

If annotated with `@readonly`, the compiler renders `@Common.FieldControl: #ReadOnly` into the edmx but in CSN we do have both. In the future, the CSN might only contain the modeled annotation without`@Common.FieldControl: #ReadOnly`.

::: warning

> You need to implement custom input validation for `@Common.FieldControl` when using static or dynamic numeric values, for example, `@Common.FieldControl: 1` or `@Common.FieldControl: integer_field`.

Reviewer: ```suggestion
Custom validations are required when using static or dynamic numeric values, for example, `@Common.FieldControl: 1` or `@Common.FieldControl: integer_field`.
```

> ### `@Common.FieldControl` {#common-fieldControl}

Reviewer 3: Hm.. let's avoid camel case in anchors.  Thats unusual.
Use `#common-fieldcontrol` and put it on the next line.  This way we get both `#common-fieldcontrol` (your anchor) and the editor's anchor `#commonfieldcontrol`

```

> ### `@Common.FieldControl` {#common-fieldControl}

Reviewer: I'd rather add that after `@mandatory/readonly` section as this belongs together.
In addition, we could add that as a warning that numeric and dynamic values are not evaluated by the runtime.

The input validation for `@Common.FieldControl: #Mandatory` and `@Common.FieldControl: #ReadOnly` is done from the CAP runtimes automatically.

> The input validation for `@Common.FieldControl` with static or dynamic numeric values(e.g. `@Common.FieldControl: 1` or `@Common.FieldControl: integer_field` ) has to be done by custom implementation.

Reviewer 2: ```suggestion
You need to implement custom input validation for `@Common.FieldControl` when using static or dynamic numeric values, for example, `@Common.FieldControl: 1` or `@Common.FieldControl: integer_field`.
```
